judges/pull-was-merged/pull-was-merged.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+# SPDX-FileCopyrightText: Copyright (c) 2024-2025 Zerocracy
+# SPDX-License-Identifier: MIT
+
+# Judge that monitors pulls which were closed or merged.
+
+require 'fbe/conclude'
+require 'fbe/octo'
+require 'fbe/github_graph'
+require 'fbe/who'
+require 'fbe/issue'
+require 'fbe/delete'
+require 'fbe/overwrite'
+
+Fbe.conclude do
+  quota_aware
+  on "(and (eq where 'github') (eq what 'pull-was-opened'))"
+
+  def self.fill_fact_by_hash(fact, hash)
+    hash.each do |prop, value|
+      fact.send(:"#{prop}=", value)
+    end
+  end
+
+  def self.comments_info(pr)
+    code_comments = Fbe.octo.pull_request_comments(pr[:base][:repo][:full_name], pr[:number])
+    issue_comments = Fbe.octo.issue_comments(pr[:base][:repo][:full_name], pr[:number])
+    {
+      comments: pr[:comments] + pr[:review_comments],
+      comments_to_code: code_comments.count,
+      comments_by_author: code_comments.count { |comment| comment[:user][:id] == pr[:user][:id] } +
+        issue_comments.count { |comment| comment[:user][:id] == pr[:user][:id] },
+      comments_by_reviewers: code_comments.count { |comment| comment[:user][:id] != pr[:user][:id] } +
+        issue_comments.count { |comment| comment[:user][:id] != pr[:user][:id] },
+      comments_appreciated: count_appreciated_comments(pr, issue_comments, code_comments),
+      comments_resolved: Fbe.github_graph.resolved_conversations(
+        pr[:base][:repo][:full_name].split('/').first, pr[:base][:repo][:name], pr[:number]
+      ).count
+    }
+  end
+
+  def self.count_appreciated_comments(pr, issue_comments, code_comments)
+    issue_appreciations =
+      issue_comments.sum do |comment|
+        Fbe.octo.issue_comment_reactions(pr[:base][:repo][:full_name], comment[:id])
+           .count { |reaction| reaction[:user][:id] != comment[:user][:id] }
+      end
+    code_appreciations =
+      code_comments.sum do |comment|
+        Fbe.octo.pull_request_review_comment_reactions(pr[:base][:repo][:full_name], comment[:id])
+           .count { |reaction| reaction[:user][:id] != comment[:user][:id] }
+      end
+    issue_appreciations + code_appreciations
+  end
+
+  def self.fetch_workflows(pr)
```

Comment:
@Yegorov this looks like code duplication with `github-events`. Can't we introduce a new class/function in the `lib/` directory, that will do this?

Reply:
@yegor256 yes, i extracted shared methods from `github-events` judge